tools/c7n_azure/c7n_azure/output.py

```diff
@@ -95,3 +109,81 @@ def get_blob_client_wrapper(output_path, ctx):
         # provides easier test isolation
         s = local_session(ctx.session_factory)
         return StorageUtilities.get_blob_client_by_uri(output_path, s)
+
+
+@metrics_outputs.register('azure')
+class MetricsOutput(Metrics):
+    """Send metrics data to app insights
+    """
+
+    def __init__(self, ctx, config=None):
+        super(MetricsOutput, self).__init__(ctx, config)
+        self.instrumentation_key = AppInsightsHelper.get_instrumentation_key(config['url'])
+        self.namespace = self.ctx.policy.name
+        self.subscription_id = local_session(self.ctx.policy.session_factory).get_subscription_id()
+        self.tc = TelemetryClient(self.instrumentation_key)
+
+    def _format_metric(self, key, value, unit, dimensions):
+        d = {
+            'Name': key,
+            'Value': value,
+            'Dimensions': {
+                'Policy': self.ctx.policy.name,
+                'ResType': self.ctx.policy.resource_type,
+                'SubscriptionId': self.subscription_id,
+                'ExecutionId': executionId,
+                'Unit': unit
+            }
+        }
+        for k, v in dimensions.items():
+            d['Dimensions'][k] = v
+        return d
+
+    def _put_metrics(self, ns, metrics):
+        for m in metrics:
+            self.tc.track_metric(name=m['Name'],
+                                 value=m['Value'],
+                                 properties=m['Dimensions'])
+        self.tc.flush()
+
+
+class AppInsightsLogHandler(LoggingHandler):
+    def __init__(self, instrumentation_key, policy, subscription_id):
+        super(AppInsightsLogHandler, self).__init__(instrumentation_key)
+        self.policy = policy
```

Comment:
nit: can you rename to policy_name just so that its clear that its a string not an object.